src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -248,6 +186,128 @@ void ElementView::keyPressEvent(QKeyEvent * event)
+        opt0.state |= QStyle::State_Enabled;
+        opt1.state |= QStyle::State_Enabled;
+        opt2.state |= QStyle::State_Enabled;
+        opt3.state |= QStyle::State_Enabled;
+
+        opt0.rect.setRect(x0 + border, btny, buttonSize, buttonSize);
+        opt1.rect.setRect(x0 + height + border, btny, buttonSize, buttonSize);
+        opt2.rect.setRect(x0 + height * 2 + border, btny, buttonSize, buttonSize);
+        opt3.rect.setRect(x0 + height * 3 + border, btny, buttonSize, buttonSize);
+
+        opt0.icon = itemData.icon0;
+        opt1.icon = itemData.icon1;
+        opt2.icon = itemData.icon2;
+        opt3.icon = itemData.icon3;
+        QSize btnSize = QSize(16, 16);
+        opt0.iconSize = btnSize; opt1.iconSize = btnSize; opt2.iconSize = btnSize; opt3.iconSize = btnSize;
+
+        QApplication::style()->drawControl(QStyle::CE_PushButton, &opt0, painter, 0);
+        QApplication::style()->drawControl(QStyle::CE_PushButton, &opt1, painter, 0);
+        QApplication::style()->drawControl(QStyle::CE_PushButton, &opt2, painter, 0);
+        QApplication::style()->drawControl(QStyle::CE_PushButton, &opt3, painter, 0);*/
+
+        //Label : 
+        painter->drawText(x0 + height * 4 + 3 * border, option.rect.y() + height - 5, itemData->label);
+
+    }
+}
+
+bool MyDelegate::editorEvent(QEvent* event, QAbstractItemModel* model, const QStyleOptionViewItem& option, const QModelIndex& index)
+{
+    if (event->type() == QEvent::MouseMove) {
+        /*if (index != m_lastUnderMouse) {
+            if (m_lastUnderMouse.isValid()) {
+                model->setData(m_lastUnderMouse, (int)Normal, Qt::UserRole);
+                emit needsUpdate(m_lastUnderMouse);
+            }
+            if (index.isValid() && index.column() == ButtonColumn) {
+                model->setData(index, (int)Hovered, Qt::UserRole);
+                emit needsUpdate(index);
+                m_lastUnderMouse = index;
+            }
+            else {
+                m_lastUnderMouse = QModelIndex();
+            }
+        }*/
+    }
+    if (event->type() == QEvent::MouseButtonPress || event->type() == QEvent::MouseButtonDblClick) {
+        QMouseEvent* mEvent = (QMouseEvent*)event;
+        QPoint point = mEvent->pos();
+        int xPos = point.x();
+        int border = 1;
+        ElementData* itemData = qvariant_cast<ElementData*>(index.data(Qt::UserRole));
+        if(xPos < option.rect.x() + 4 + option.rect.height() + border)
+            itemData->clickedOn = 0; //0 = edge
+        else if (xPos < option.rect.x() + 4 + option.rect.height() * 2 + border)
+            itemData->clickedOn = 1; //1 = start
+        else if (xPos < option.rect.x() + 4 + option.rect.height() * 3 + border)
+            itemData->clickedOn = 2; //2 = end
+        else if (xPos < option.rect.x() + 4 + option.rect.height() * 4 + border)
+            itemData->clickedOn = 3; //3 = mid
+        else
+            itemData->clickedOn = 4; //4 = none
+
+    }

```

Comment:
Maybe could manage right-click in this function ?

Reply:
You added a `ElementView::mousePressEvent` that just manages RMB press. Maybe this code can be centralized here.

Reply:
Oh yes indeed